saleor/payment/tasks.py

```diff
@@ -27,16 +27,43 @@ def transactions_to_release_funds():
     predefined TTL. It then fetches related transactions that are authorized or charged,
     ready for fund release.
     """
-    expired_checkouts_time = (
-        datetime.now(pytz.UTC) - settings.CHECKOUT_TTL_BEFORE_RELEASING_FUNDS
+    now = datetime.now(pytz.UTC)
+    channels = (
+        Channel.objects.using(settings.DATABASE_CONNECTION_REPLICA_NAME)
+        .annotate(
+            last_transaction_modified_at=ExpressionWrapper(
+                OuterRef(
+                    "last_transaction_modified_at",
+                ),
+                output_field=DateTimeField(),
+            )
+        )
+        .filter(
+            Q(release_funds_for_expired_checkouts=True)
+            & Q(
+                checkout_ttl_before_releasing_funds__lt=now - OuterRef("last_change")  # type: ignore[operator]
+            )
+            & (
+                Q(last_transaction_modified_at__isnull=True)
+                | Q(
+                    checkout_ttl_before_releasing_funds__lt=now
+                    - OuterRef("last_transaction_modified_at")  # type: ignore[operator]
+                )
+            )
+            & Q(pk=OuterRef("channel_id"))
+            & (
+                Q(checkout_release_funds_cut_off_date__isnull=True)
+                | Q(checkout_release_funds_cut_off_date__lt=OuterRef("last_change"))
```

Comment:
IMHO, it shouldn't be `last_change`. It should be `created`. The main purpouse of these flag is telling the moment when your "old" release flow stopped working, and Saleor needs to take the responsibility. 
With `last_change` you will have mixed checkouts, created before and created after cut_off date - so you will have checkouts covered by "old" flow, and now they will be also included in the new one